### ISSUE TITLE
Indy tab shows incorrect data when running corp jobs

### ISSUE THREAD

**Issue by kunnis** [evemondevteam/evemon#29]
So if you're running both Corp Jobs & Personal jobs, and have API keys loaded for both, EveMon doesn't show the corp jobs, and at the bottom of the window it says "Manufacturing Jobs Remaining: 10 of 10"  I think should reflect the corp jobs in build.

I'm willing to fix the issue, but I'd like to know how you would like it solved.  A long time ago I tried submitting a patch that would show the corp jobs on the character window, but it wasn't merged in because I think other plans were in the work.

**Comment by JimiC** [comment]
EVEMon will not show corp jobs in the character indy monitor, unless you are a CEO or Director and provide a corp API key.
That said EVEMon, should NOT present any corp data in any character monitor, but on another corp monitor feature (which was planned to be implemented but never got done).

**Comment by JimiC** [comment]
No, it's not logical to reflect the corp indy jobs in a `character` monitor if those jobs weren't set by that character.

It's not as simple as you put it to reflect the "jobs remaining". Believe me I fought that "monster". There are too many parameters in play, and what you are discussing surely belongs to its own context (feature).

**Comment by kunnis** [comment]
Agreed, I think I wasn't being clear in my earlier statement.  I was thinking of only displaying the corp jobs you started in the character screen.

**Comment by JimiC** [comment]
This is already implemented and EVEMon also has a filtering for that (click on the "cog" icon).
This works when you provide a corp API key and any order provided from that API, match your character name.

**Comment by kunnis** [comment]
Correct.  But it doesn't work for my non-director characters.

**Comment by JimiC** [comment]
Corp API key is bind to the character that issue it. That's why.

**Comment by kunnis** [comment]
Yes, and I'd like to remove that binding.

**Comment by JimiC** [comment]
You shouldn't.

**Comment by kunnis** [comment]
Or make the binding share across all the characters?

**Comment by JimiC** [comment]
> Or make the binding share across all the characters?
On a corp dedicated feature? Sure.

**Comment by kunnis** [comment]
I was thinking install the director key => All characters in the director's cor get the corp jobs they started showing up up on their Active Jobs section.

**Comment by JimiC** [comment]
It's against EVEMon's character monitor design. If you are up to that build a Corp Management monitor and do that there.

**Comment by kunnis** [comment]
It's monitoring the jobs the character installed.  When you run a lot of indy jobs, you need to know which characters you need to login to service their jobs.  It just so happens to be in the corp API

**Comment by kunnis** [comment]
and to expand on this a bit more, I'd think if there was a corp management section, you'd still show the job on both the corp sheet (to get a corp overview) and the character's sheet (to get a view of what the character is doing)

**Comment by JimiC** [comment]
OK. Let's try this again. One character issues the Corp API key (which is the CEO or Director of the Corp), so for EVEMon's _character_ monitor only that character can see those corp indy jobs and only those that was issue by that character, because that's the point of a _character_ monitor feature.

On a Corp Management monitor, all Corp jobs should be shown, regardless if they are already shown in the character's indy monitor.

So to answer your expanded comment, yes, some indy jobs would show in both monitors.

**Comment by kunnis** [comment]
I want to change get rid of the whole issuing character requirement.  I'm in the case of being a non-director that has the corp API keys.

**Comment by JimiC** [comment]
Again, you shouldn't change that on a _character_ monitor level.

And on the EVE metadata gameplay front, no other character should be able to see your corp data, without permission. What if the Corp API key got stolen? Would you like your corp data to be exposed?

Your issue could be solved if CCP had given access to corp indy jobs also to the factory manager role.
We had requested that since the new API keys management was introduced, but...

**Comment by kunnis** [comment]
That's why I'm discussing the change with you.  I think we should change that.  View this as the first step to building the Corp elements of the app.  Decouple the Corp API key from being tied to a specific character.

**Comment by JimiC** [comment]
If and when EVEMon gets a Corp Management feature.

And to extend my previous comment, when you design a software you should not only assess what's possible to do, but also if you should do it. Especially when it comes to securing sensitive data.

**Comment by kunnis** [comment]
Why not start now?  I'm willing to go through and remove the ties between the Corp API key and a specific character.

I've also had this problem in the past.  I had an alt-corp, with 6 alts in it.  the 6 alts were doing indy work, but the corp only had one director character.  I don't want to make all the alts directors just to make evemon work.   However, I do understand "Don't promote people doing insecure things".

**Comment by JimiC** [comment]
The bind should only be lifted when the Corp Management feature is implemented. And even then that feature should be bind to Corp roles.

**Comment by kunnis** [comment]
Why should we delay lifting the bind?  Just unbind it, and on the character sheets show the corp jobs you installed.

**Comment by JimiC** [comment]
Simple because your CEO or Director didn't gave you that permission (reminder the Corp API key was issued by the CEO - Director and **only** that character should be able to see the Corp data without explicit permissions).

You should better ask CCP to extend the way Corp API keys work and add role permissions to them.

**Comment by kunnis** [comment]
The director issued the key, yes.  However, it's not for info solely about the director's character.  It's for info about the corp.
